### PR TITLE
Add KLIMA supply API

### LINKED ISSUE #198: Add current total supply API
In order for CMC to properly display our market cap on their site, we need to provide them with a specific API endpoint that just returns the total supply as a number (and nothing else)

Tried to direct them to use our Subgraph but they don't seem to be that flexible and I can't figure out a way to get a single value out of a GraphQL URL-based query

So we can just create a simple endpoint like `/supply` that fetches the current total supply from the KLIMA contract ABI and returns that single value.

Original request received from CMC:
image

### PR DESCRIPTION
**Pull request by 0xAurelius**
## Description
Adds API endpoint for current KLIMA supply

## Related Ticket

Resolves #198 

## Checklist

- [ ] Building the site with `npm run build-site` works without errors
- [ ] Building the app with `npm run build-app` works without errors
- [ ] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

**Comment by GreenGauss** [comment]
nice will take a closer look soon, but lets put all /api/ endpoints under our main domain not the dapp

in other words if you put it in `site` instead of `dapp` repo then the endpoint will be:
```
klimadao.finance/api/klima-supply/
```

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
Nice! Let's move this to `site` instead of `dapp` so all api endpoints are at our root domain `klimadao.finance/api/klima-supply`

File: app/pages/api/klima-supply/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { NextApiRequest, NextApiResponse } from "next/types";
+import { getKlimaSupply } from "@klimadao/lib/utils";
+
+type ResponseData = {
+    currentKlimaSupply?: string;
+    error?: string;
+};
+
+export default async function handler(
```
Comment (Atmosfearful) [inline app/pages/api/klima-supply/index.ts:9]: We can improve the typings a bit:
```suggestion
interface Response {
  currentKlimaSupply: string;
};

const handler: NextApiHandler<Response> = async (req, res) => {
```

**Review by Atmosfearful — CHANGES_REQUESTED** [review]

File: app/pages/api/klima-supply/index.ts (hunk 1 above)
Comment (0xAurelius) [reply app/pages/api/klima-supply/index.ts]: I realized this API actually needs to be simplified to meet CMC's requirement of just returning the supply and nothing else

**Comment by 0xAurelius** [comment]
OK moved it to `site` and simplified things to only return the supply itself and no JSON wrapping

**Comment by 0xAurelius** [comment]
Finally got the build working, API endpoint seems to work in preview deployment: https://klimadao-site-git-fork-0xaurelius-supply-api-klimadao.vercel.app/api/klima-supply

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**